### PR TITLE
Fix warnings: 1822, 1823, 1825

### PR DESCRIPTION
**Pull request by azabbasi**
There are still 1822 and 1823 warnings left in the project that all belong to the following files:
Fx.cs
ExceptionTrace.cs

I have decided to address these two files in a separate PR as they are cluttered with warnings and unused code. That will be coming up shortly.

### CODE REVIEW

File: iothub/device/src/AmqpConnectionPoolSettings.cs [hunk 1]
```diff
@@ -6,13 +6,12 @@
 namespace Microsoft.Azure.Devices.Client
 {
     /// <summary>
-    /// contains Amqp Connection Pool settings for DeviceClient
+    /// contains AMQP Connection Pool settings for DeviceClient
     /// </summary>
     public sealed class AmqpConnectionPoolSettings
     {
-        private static readonly TimeSpan s_defaultConnectionIdleTimeout = TimeSpan.FromMinutes(2);
```
Comment (azabbasi) [inline iothub/device/src/AmqpConnectionPoolSettings.cs:13]: not used

File: iothub/device/src/Edge/NullCertificateValidator.cs [hunk 2]
```diff
@@ -11,11 +11,6 @@ internal class NullCertificateValidator : ICertificateValidator
     {
         public static NullCertificateValidator Instance { get; } = new NullCertificateValidator();
 
-        public void SetupCertificateValidation()
```
Comment (azabbasi) [inline iothub/device/src/Edge/NullCertificateValidator.cs:14]: not used

File: iothub/device/src/HsmAuthentication/HttpHsmSignatureProvider.cs [hunk 3]
```diff
@@ -18,18 +18,14 @@ namespace Microsoft.Azure.Devices.Client.HsmAuthentication
 {
     internal class HttpHsmSignatureProvider : ISignatureProvider
     {
-        private const string DefaultApiVersion = "2018-06-28";
```
Comment (azabbasi) [inline iothub/device/src/HsmAuthentication/HttpHsmSignatureProvider.cs:21]: not used

File: iothub/device/src/net451/IotHubClientWebSocket.cs [hunk 4]
```diff
@@ -1010,29 +1005,9 @@ public static int IndexOfAsciiChars(byte[] array, int offset, int count, char as
             return -1;
         }
 
-        protected static bool InitializeDisableServerCertificateValidation()
```
Comment (azabbasi) [inline iothub/device/src/net451/IotHubClientWebSocket.cs:1013]: always false

File: iothub/device/src/net451/IotHubClientWebSocket.cs [hunk 5]
```diff
@@ -1010,29 +1005,9 @@ public static int IndexOfAsciiChars(byte[] array, int offset, int count, char as
...
-        }
-
         public static bool OnRemoteCertificateValidation(object sender, X509Certificate certificate, X509Chain chain, SslPolicyErrors sslPolicyErrors)
         {
-            if (sslPolicyErrors == SslPolicyErrors.None)
-            {
-                return true;
-            }
-
-            if (s_disableServerCertificateValidation.Value && sslPolicyErrors == SslPolicyErrors.RemoteCertificateNameMismatch)
```
Comment (azabbasi) [inline iothub/device/src/net451/IotHubClientWebSocket.cs:1030]: always false

File: iothub/device/src/net451/IotHubClientWebSocket.cs [hunk 6]
```diff
@@ -60,9 +58,6 @@ internal class IotHubClientWebSocket
         private readonly string _requestPath;
         private string _webSocketKey;
         private string _host;
-        private const string DisableServerCertificateValidationKeyName = "Microsoft.Azure.Devices.DisableServerCertificateValidation";
-
-        private static readonly Lazy<bool> s_disableServerCertificateValidation = new Lazy<bool>(InitializeDisableServerCertificateValidation);
```
Comment (azabbasi) [inline iothub/device/src/net451/IotHubClientWebSocket.cs:65]: always false so removed

File: iothub/device/src/net451/IotHubClientWebSocket.cs [hunk 7]
```diff
@@ -60,9 +58,6 @@ internal class IotHubClientWebSocket
         private readonly string _requestPath;
         private string _webSocketKey;
         private string _host;
-        private const string DisableServerCertificateValidationKeyName = "Microsoft.Azure.Devices.DisableServerCertificateValidation";
```
Comment (azabbasi) [inline iothub/device/src/net451/IotHubClientWebSocket.cs:63]: not used

File: iothub/device/src/Common/Extensions/ExceptionExtensions.cs [hunk 8]
```diff
@@ -50,6 +50,10 @@ public static IEnumerable<TException> Unwind<TException>(this Exception exceptio
             return exception.Unwind().OfType<TException>();
         }
 
+        [System.Diagnostics.CodeAnalysis.SuppressMessage(
+            "Performance",
+            "CA1825:Avoid zero-length array allocations",
+            Justification = "net451 does not support Array.Empty<T>")]
```
Comment (drwill-ms) [inline iothub/device/src/Common/Extensions/ExceptionExtensions.cs:56]: Or you can do as we have elsewhere, where you have an `#if NET451` directive and change the type based on framework. Then later when we deprecate net451, we just remove the relevant directives.

**Review by drwill-ms — APPROVED** [review]
Nice job! :)

File: iothub/device/src/Common/Extensions/ExceptionExtensions.cs (hunk 8 above)
Comment (azabbasi) [reply iothub/device/src/Common/Extensions/ExceptionExtensions.cs]: done.

File: iothub/device/src/AmqpConnectionPoolSettings.cs [hunk 9]
```diff
@@ -6,13 +6,12 @@
 namespace Microsoft.Azure.Devices.Client
 {
     /// <summary>
-    /// contains Amqp Connection Pool settings for DeviceClient
+    /// contains AMQP Connection Pool settings for DeviceClient
```
Comment (abhipsaMisra) [inline iothub/device/src/AmqpConnectionPoolSettings.cs:9]: nit: C caps! :D

File: iothub/device/src/Edge/NullCertificateValidator.cs (hunk 2 above)
Comment (abhipsaMisra) [reply iothub/device/src/Edge/NullCertificateValidator.cs:14]: is the class public?

File: iothub/device/src/HsmAuthentication/HttpHsmSignatureProvider.cs [hunk 10]
```diff
@@ -99,10 +95,10 @@ public async Task<string> SignAsync(string moduleId, string generationId, string
             }
         }
 
-        private async Task<SignResponse> SignAsyncWithRetry(HttpHsmClient hsmHttpClient, string moduleId, string generationId, SignRequest signRequest)
+        private async Task<SignResponse> SignAsyncWithRetryAsync(HttpHsmClient hsmHttpClient, string moduleId, string generationId, SignRequest signRequest)
         {
-            var transientRetryPolicy = new RetryPolicy(TransientErrorDetectionStrategy, TransientRetryStrategy);
-            SignResponse response = await transientRetryPolicy.ExecuteAsync(() => hsmHttpClient.SignAsync(_apiVersion, moduleId, generationId, signRequest));
+            var transientRetryPolicy = new RetryPolicy(s_transientErrorDetectionStrategy, s_transientRetryStrategy);
+            SignResponse response = await transientRetryPolicy.ExecuteAsync(() => hsmHttpClient.SignAsync(_apiVersion, moduleId, generationId, signRequest)).ConfigureAwait(false);
```
Comment (abhipsaMisra) [inline iothub/device/src/HsmAuthentication/HttpHsmSignatureProvider.cs:101]: whoa, configure await was missing!

**Review by abhipsaMisra — APPROVED** [review]

File: iothub/device/src/AmqpConnectionPoolSettings.cs (hunk 9 above)
Comment (azabbasi) [reply iothub/device/src/AmqpConnectionPoolSettings.cs:9]: I will change that in a following PR since I already ran the gates

**Pull request merged**